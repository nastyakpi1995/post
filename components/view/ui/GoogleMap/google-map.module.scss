.map {
  display: block;
}

.map_text {
  display: block;
}