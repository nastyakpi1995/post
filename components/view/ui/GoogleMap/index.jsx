import React from 'react';
import PropTypes from 'prop-types';
import GoogleMapReact from 'google-map-react';
import styles from './google-map.module.scss';

const Marker = ({ text }) => <div>{text}</div>;

const GoogleMap = ({ height, width, lat, lng, zoom, text }) => {
  const center = {
    lat,
    lng,
  };

  return (
    <div
      style={{ height: `${height}`, width: `${width}` }}
      className={styles.map}
    >
      <GoogleMapReact
        bootstrapURLKeys={{ key: process.env.GOOGLE_MAPS_KEY }}
        defaultCenter={center}
        defaultZoom={zoom}
      >
        <Marker lat={lat} lng={lng} text={text} />
      </GoogleMapReact>
    </div>
  );
};

Marker.propTypes = {
  text: PropTypes.string,
};

GoogleMap.defaultProps = {
  zoom: 11,
  lat: 59.955413,
  lng: 30.337844,
  height: '100vh',
  width: '100%',
  text: 'My location',
};

GoogleMap.propTypes = {
  text: PropTypes.string,
  zoom: PropTypes.number,
  lat: PropTypes.number,
  lng: PropTypes.number,
  height: PropTypes.string,
  width: PropTypes.string,
};

export default GoogleMap;
