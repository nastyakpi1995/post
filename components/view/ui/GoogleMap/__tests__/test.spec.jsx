import React from 'react';
import renderer from 'react-test-renderer';
import GoogleMap from '../index';

it('renders correctly', () => {
  const component = renderer.create(<GoogleMap />).toJSON();
  expect(component).toMatchSnapshot();
});
