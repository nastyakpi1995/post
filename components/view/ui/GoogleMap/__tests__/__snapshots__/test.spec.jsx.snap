// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <div
    onClick={[Function]}
    onMouseDownCapture={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "height": "100%",
        "margin": 0,
        "padding": 0,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "left": 0,
          "margin": 0,
          "padding": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
    <div
      style={
        Object {
          "height": "50%",
          "left": "50%",
          "margin": 0,
          "padding": 0,
          "position": "absolute",
          "top": "50%",
          "width": "50%",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "left": 0,
            "margin": 0,
            "padding": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "transparent",
              "height": 0,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 0,
            }
          }
        >
          <div>
            My location
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
